go/store/datas/database_common.go

```diff
@@ -184,13 +210,13 @@ func (db *database) doSetHead(ctx context.Context, ds Dataset, newHeadRef types.
 		return err
 	}
 
-	commitRef, err := db.WriteValue(ctx, commit) // will be orphaned if the tryCommitChunks() below fails
```

Comment:
would call this refSt instead of stRef as we use that st as a suffix all over this function.